graders/c/cgrader/cgrader.py

```diff
@@ -73,13 +74,36 @@ def __init__(self):
 UngradableException = UngradableError
 
 
+def is_str_list(val: list[float | str | int]) -> TypeGuard[list[str]]:
+    """Determines whether all objects in the list are strings"""
+    return all(isinstance(x, str) for x in val)
+
+
+# TODO: in version 3.11 we can use `Required` to mark properties as such.
```

Comment:
FWIW, this grader already runs on Python 3.12:

```
$ docker run --rm -it prairielearn/grader-c python3 --version
Python 3.12.3
```

Reply:
Ah -- good to know. I didn't catch that since my editor is using the PrairieLearn Python. Potentially this can wait till #10984